examples/IB/explicit/ex7/example.cpp

```diff
@@ -0,0 +1,435 @@
+                    "Q_bc_coefs", app_initializer->getComponentDatabase("ConcentrationBcCoefs"), grid_geometry));
+                adv_diff_integrator->setPhysicalBcCoef(Q_var, Q_bc_coefs.get());
+            }
+        }
+
+        // Initialize hierarchy configuration and data on all patches.
+        time_integrator->initializePatchHierarchy(patch_hierarchy, gridding_algorithm);
+
+        // Deallocate initialization objects.
+        ib_method_ops->freeLInitStrategy();
+        ib_initializer.setNull();
+        app_initializer.setNull();
+
+        // Print the input database contents to the log file.
+        plog << "Input database:\n";
+        input_db->printClassData(plog);
+
+        // Write restart data before starting main time integration loop.
+        if (dump_restart_data && !is_from_restart)
+        {
+            pout << "\nWriting restart files...\n\n";
+            RestartManager::getManager()->writeRestartFile(restart_dump_dirname, 0);
+        }
+        // Write out initial visualization data.
+        int iteration_num = time_integrator->getIntegratorStep();
+        double loop_time = time_integrator->getIntegratorTime();
+        if (dump_viz_data && uses_visit)
+        {
+            pout << "\n\nWriting visualization files...\n\n";
+            time_integrator->setupPlotData();
+            visit_data_writer->writePlotData(patch_hierarchy, iteration_num, loop_time);
+            silo_data_writer->writePlotData(iteration_num, loop_time);
+        }
+        
+        // Create Text File which Stores the locations of each IB point
+
+        ofstream fout("Coordinates.txt", ios::out);
+
+        // Main time step loop.
+        double loop_time_end = time_integrator->getEndTime();
+        double dt = 0.0;
+        while (!MathUtilities<double>::equalEps(loop_time, loop_time_end) && time_integrator->stepsRemaining())
+        {
+            iteration_num = time_integrator->getIntegratorStep();
+            loop_time = time_integrator->getIntegratorTime();
+
+            pout << "\n";
+            pout << "+++++++++++++++++++++++++++++++++++++++++++++++++++\n";
+            pout << "At beginning of timestep # " << iteration_num << "\n";
+            pout << "Simulation time is " << loop_time << "\n";
+
+
+            update_target_points(patch_hierarchy, ib_method_ops->getLDataManager(), loop_time, dt);
+
+
+            dt = time_integrator->getMaximumTimeStepSize();
+            time_integrator->advanceHierarchy(dt);
+            print_eul_data(patch_hierarchy, ib_method_ops->getLDataManager());
+            loop_time += dt;
+
+            pout << "\n";
+            pout << "At end       of timestep # " << iteration_num << "\n";
+            pout << "Simulation time is " << loop_time << "\n";
+            pout << "+++++++++++++++++++++++++++++++++++++++++++++++++++\n";
+            pout << "\n";
+
+            std::ofstream fout("Coordinates.txt", std::ios::app);
+
```

Comment:
We already created the output stream above:
```suggestion
```
In addition, this isn't going to work in parallel - each process will attempt to create the same text file and the program will crash. You will need to decide if we want to use one coordinate file for each process or one giant coordinate file that saves things from each process.